### PR TITLE
Switch to critest

### PR DESCRIPTION
**Pull request by heartlock**
Signed-off-by: heartlock <21521209@zju.edu.cn>
Closes: #143 

- [x] Step1: enable critest in travis CI
- [x] Step2: delete reduplicate test case in the old e2e test framwork, only keep frakti-specific tests

@feiskyer But critest still test fail with frakti. @xlgao-zju Please read travis CI log.

### CODE REVIEW

**Comment by feiskyer** [comment]
Yep, image related tests are failed because

```
RemoveImage from hyper runtime service failed: rpc error: code = 2 desc = conflict: unable to delete c75bebcdd211 (must be forced) - image is referenced in one or more repositories
```

This is a issue caused by hyperd, which doesn't accept removing a image via id:

```
# hyperctl rmi 3448f27c273f
Error remove the image(3448f27c273f): Error from daemon's response: conflict: unable to delete 3448f27c273f (must be forced) - image is referenced in one or more repositories
```

**Comment by heartlock** [comment]
@feiskyer We can send `-f` to hyperd to fix this issue. What do you think?

**Comment by feiskyer** [comment]
let's add force in hyperclient

**Comment by heartlock** [comment]
@feiskyer List image with digest still fail.

**Comment by feiskyer** [comment]
weird, it success on my local environment. Could you add a log at [here](https://github.com/kubernetes/frakti/blob/master/pkg/manager/manager.go#L493) and run again?

**Comment by heartlock** [comment]
I add log and run again, list image return nothing. Do you test with newest version hyperd?
`I0519 03:57:02.934697   22272 manager.go:493] List images response []`

**Comment by feiskyer** [comment]
I'm testing it with hyperd 0.8.1, not master version.

**Comment by feiskyer** [comment]
Update: the test also pass on hyperd master.

**Comment by feiskyer** [comment]
@heartlock Passed at #148. It upgraded docker to v1.13.

**Comment by heartlock** [comment]
ok, will do it.

**Comment by xlgao-zju** [comment]
This may be caused by [kubernetes/kubernetes#43267](https://github.com/kubernetes/kubernetes/issues/43267) and the mixed-runtime.

**Comment by feiskyer** [comment]
@xlgao-zju Right, so we are running with docker 1.13 in cri-tools.

**Comment by feiskyer** [comment]
Some security contexts are not supported, we should skip those tests

**Comment by feiskyer** [comment]
port mapping with host port and container port is also not supported yet (see #122 )

**Comment by feiskyer** [comment]
@heartlock any updates on this?

**Comment by heartlock** [comment]
@feiskyer  Attach container TEST case block travis-ci so that test job exceeded the maximum time limit for jobs, and has been terminated. I do not know what the reason is because. It seems something wrong  with hyperd. Please have a look.

File: hack/test-e2e.sh [hunk 1]
```diff
@@ -154,7 +154,7 @@ runTests() {
 
   start_frakti
 
-  frakti::test::e2e
+  #frakti::test::e2e
```
Comment (feiskyer) [inline hack/test-e2e.sh]: Let's keep this now and remove e2e tests dup with critest.

**Comment by feiskyer** [comment]
Filed https://github.com/hyperhq/hyperd/pull/637 to fix attach problem.

**Comment by heartlock** [comment]
@feiskyer Ready for review!

**Comment by feiskyer** [comment]
@heartlock Cool

**Pull request merged**